core/state_processor.go

```diff
@@ -167,12 +178,18 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
 	blockContext := NewEVMBlockContext(header, bc, author)
 	txContext := NewEVMTxContext(msg)
 	vmenv := vm.NewEVM(blockContext, txContext, statedb, config, cfg)
-	return applyTransaction(msg, config, gp, statedb, header.Number, header.Hash(), tx, usedGas, vmenv)
+	return ApplyTransactionWithEVM(msg, config, gp, statedb, header.Number, header.Hash(), tx, usedGas, vmenv)
 }
 
 // ProcessBeaconBlockRoot applies the EIP-4788 system call to the beacon block root
 // contract. This method is exported to be used in tests.
-func ProcessBeaconBlockRoot(beaconRoot common.Hash, vmenv *vm.EVM, statedb *state.StateDB) {
+func ProcessBeaconBlockRoot(beaconRoot common.Hash, vmenv *vm.EVM, statedb *state.StateDB, logger BlockchainLogger) {
+	if logger != nil {
+		logger.OnBeaconBlockRootStart(beaconRoot)
+		defer func() {
+			logger.OnBeaconBlockRootEnd()
```

Comment:
```suggestion
		logger.OnSystemCallStart(beaconRoot)
		defer func() {
			logger.OnSystemCallEnd()
```

Reply:
instead of `logger BlockchainLogger` , perhaps use `evm.logger` ?

Reply:
That's the name `SystemCall` we have used in our block model for this instrumentation. I personally like having a system call "handler" than the more specific `OnBeaconBlockRoot`.

For blockchain vs EVM, I think `BlockchainLogger` makes more sense. The operation of doing this specific action is part of the blockchain rules and not the EVM rules directly.